### PR TITLE
fix: hmr schema watcher had an always truthy conditional check

### PR DESCRIPTION
**Pull request by sduduzog**
Multimatch always returns an array, even an empty one when there's no match, which is a truthy value. This meant that templates were always updated even if it's not schema files that were changed.

Another significant change is resolving schema as path patterns. The `path` argument from the watch hook is absolute, so an unresolved schema that starts with ./ would have always been false. This would not have been found because of the first bug with the matcher condition.

I'm not sure, but it might help with https://github.com/tobiasdiez/nuxt-graphql-server/issues/57

### CODE REVIEW

**Comment by tobiasdiez** [comment]
Thanks for your contributions. On a first glance they look good, but I will only have time for a more in depth review in a few weeks once I'm back from holidays. Sorry for the delay

**Comment by sduduzog** [comment]
Oh no, no worries. Enjoy the rest of your holidays @tobiasdiez I don't think there's much urgency for this at this time

File: src/module.ts [hunk 1]
```diff
@@ -103,14 +105,19 @@ export default defineNuxtModule<ModuleOptions>({
     if (nuxt.options.dev) {
       nuxt.hook('nitro:build:before', (nitro) => {
         nuxt.hook('builder:watch', async (event, path) => {
-          if (multimatch(path, options.schema)) {
+           const schema = Array.isArray(options.schema)
```
Comment (tobiasdiez) [inline src/module.ts:108]: This doesn't work for me in the playground. There `schema` is now resolved to `absolute/path/server/**/*.graphql` while the changed `path` reported by nuxt is `server/schema.graphql`. 
This is in contrast to what you write above: "The path argument from the watch hook is absolute". Maybe that's a difference between os? I'm on windows....

**Review by tobiasdiez — COMMENTED** [review]
I've now played around with your branch, but have issues with the schema resolving. For me `path` is not an absolute path...

File: src/module.ts (hunk 1 above)
Comment (tobiasdiez) [reply src/module.ts:108]: Looking at https://github.com/nuxt/nuxt/blob/559a72370b12dc05f9d0d225e46fafd299695bcc/packages/nuxt/src/imports/module.ts#L67-L70, it seems like the builder:watch hook is indeed returning a relative path.

Comment (tobiasdiez) [reply src/module.ts:108]: After a bit more digging I've found https://github.com/nuxt/nuxt/pull/22333. So apparently it depends on the nuxt version. So if you add `path = resolve(nuxt.options.srcDir, relativePath)` then it should work in all cases (and change `rootDir` below to `srcDir`).

Comment (tobiasdiez) [reply src/module.ts]: Nevermind, I quickly did this myself.

**Comment by tobiasdiez** [comment]
Thanks again for your contribution, very much appreciated!

**Pull request merged**